weave/trace_server/validation_util.py

```diff
@@ -0,0 +1,55 @@
+import base64
+import typing
+import uuid
+
+from . import refs_internal
+
+
+def require_uuid(s: str) -> str:
+    lower_s = s.lower()
+    if lower_s != s:
+        raise ValueError(f"UUIDs must be lowercase: {s}")
+
+    try:
+        s_prime = str(uuid.UUID(s))
+    except ValueError:
+        raise ValueError(f"Invalid UUID: {s}")
+
+    if s_prime != s:
+        raise ValueError(f"Invalid UUID: {s}")
```

Comment:
are these errors the same for some security reason? might be nice to keep them separate, in case there is a bug that somehow only affects the UUID constructor for example.

Reply:
Nice - i just updated each error to be unique